spec/bundler/plugin/index_spec.rb

```diff
@@ -105,6 +105,39 @@
     end
   end
 
+  describe "remove plugin" do
+    let(:plugin_path) { lib_path("new-plugin").to_s }
+    before do
+      index.remove_plugin_paths("new-plugin")
```

Comment:
Also, I can't find where you are registering the plugin with the index?

Instead of creating `:plugin_path` just override the `:sources`, `:commands` and `:hooks` with some values and then try removing it.